File: 8_kyu/They_say_that_only_the_name_is_long_enough_to_attract_attention_They_also_said_that_only_a_simple_Kata_will_have_someone_to_solve_it_This_is_a_sadly_story_1_Are_they_opposite.c
#include <stddef.h>
#include <stdbool.h>

// Returns true if both strings are equal but case is opposite.
// Supports only standard ASCII charset letters. 
bool is_opposite(const char s1[], const char s2[]) {
  // NULL and empty strings exit
  if( s1 == NULL || s2 == NULL || *s1 == 0 || *s2 == 0 )
    return false;
  // Check char by char.
  size_t i = 0;
  while( s1[i] != 0 && s2[i] != 0 ) {
    // Quick exit if not letter case bit is the only difference.
    // ONLY works for standard ASCII letters.
    if( (s1[i] ^ s2[i]) != 0x20 )
      return false;
    i++;
  }
  // ok if end of string reached for both strings.
  return s1[i] == s2[i];
}

File: 8_kyu/They_say_that_only_the_name_is_long_enough_to_attract_attention_They_also_said_that_only_a_simple_Kata_will_have_someone_to_solve_it_This_is_a_sadly_story_1_Are_they_opposite.py
def is_opposite(s1 :str, s2 :str) -> bool:
    """
    Checks is strings have same letters but opposite case.
    WARNING: Does not support special letters like  "β".upper() == "SS".
    """
    if not s1 or not s2:
        # Special case empty string
        return False
    return s1 == s2.swapcase()
